src/rust/engine/src/nodes.rs

```diff
@@ -198,6 +156,21 @@ impl Select {
       return future::ok(value.clone()).to_boxed();
     }
 
+    fn select(select: &Select, context: &Context, product: TypeConstraint) -> NodeFuture<Value> {
```

Comment:
Should this take the `Select` node as `self`?

Reply:
Promoted to struct-level fn, done